[Documentation] DQT module Readme

knoll-alex opened the pull request:
## Brief summary of changes

This PR restores and updates a module spec for the DQT that was accidentally deleted in PR [#3437](https://github.com/aces/Loris/pull/3437) 

In addition, I deleted the readme.md file that was mistakenly stored in the help folder. This is a redo of PR #5219 which wasn't created on my fork. 

#### Testing instructions

1. Please review content, specifically Config section, to ensure info is up to date

#### Links to related tickets 

* Originally reported in Issue #5196 
* Bug source: accidental delete in PR #3437
* https://github.com/aces/Loris/projects/6

johnsaigle commented on modules/dataquery/readme/readme.md:
```diff
@@ -0,0 +1,78 @@
+`cd $lorisroot/tools
+
+#Import the base candidate data
+php CouchDB_Import_Demographics.php
```
It would be good to clarify that these import scripts are found under `tools/`

johnsaigle commented on modules/dataquery/readme/readme.md:
```diff
@@ -0,0 +1,78 @@
+</CouchDB>`
+
+To load the Data Query Tool with data stored in LORIS, run the `CouchDB_Import_*` scripts, in tools/ directory: 
+`cd $lorisroot/tools
```
```suggestion
`cd $lorisroot/tools`
```

johnsaigle commented on modules/dataquery/readme/readme.md:
```diff
@@ -0,0 +1,78 @@
+To load the Data Query Tool with data stored in LORIS, run the `CouchDB_Import_*` scripts, in tools/ directory: 
+`cd $lorisroot/tools
+
+#Import the base candidate data
```
I think the rendering for this formatting is not working properly. I think you need a space after the `#` and maybe a new line between the title and the text.

johnsaigle commented:
HI @knoll-alex, unfortunately it seems that the extra binary file wasn't removed during your latest commit

driusan commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,78 @@
+..* i.e.
+`curl -H 'Content-Type: application/json' -X POST http://$YOURCOUCHDBADMIN:$YOURCOUCHADMINPASS@$YOURSERVERNAME:5984/_replicate -d '{"source":"http://couchdb.loris.ca:5984/dataquerytool-1_0_0", "target":"$YOURDATABASENAME"}'`
+
+#### The Hard Way (for developers)
```
I don't think "The Hard Way" should be here. It's in the README of the aces/Data-Query-Tool repo, and only required for developers. The code that you need to know how to do that for also hasn't been touched in years.

knoll-alex replied:
deleted!

driusan commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,78 @@
+Use of the DQT module requires setting up the CouchDB server and credentials:
+
+[CouchDB](http://couchdb.apache.org)
+[Erica](https://github.com/benoitc/erica)
```
Erica isn't required if the hard way is removed

knoll-alex replied:
deleted!

johnsaigle commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,64 @@
+    <adminpass>adminpass</adminpass>
+</CouchDB>`
+
+To load the Data Query Tool with data stored in LORIS, run the `CouchDB_Import_*` scripts, in tools/ directory: 
```
```suggestion
To load the Data Query Tool with data stored in LORIS, run the `CouchDB_Import_*` scripts, in tools/ directory: 

```

driusan commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,64 @@
+
+[CouchDB](http://couchdb.apache.org)
+
+Note: Your version of Erlang should be the most recent version that is compatible with both your CouchDB.
```
`that is compatible with both your CouchDB` isn't grammatically correct.. I think we can probably just remove this note.

christinerogers replied:
👍  vote to remove this line

knoll-alex replied:
Removed

knoll-alex commented:
I believe - hope - that I fixed the rendering issue @johnsaigle

christinerogers commented:
> I believe - hope - that I fixed the rendering issue @johnsaigle

Hey alex - my earlier commits (`fixing snippits`) today fixed the rendering, per my slack message.  
Let's look at sorting out these most recent changes on Thursday.

christinerogers commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,79 @@
+The most closely related module to the DQT is the Data Release module, which is used to store snapshots of data for any given release. 
+
+## Permissions
+Users must have `dataquery_view` permission to use the DQT. It’s important to note that more granular permissions do not exist for this module, to restrict access to a subset of the data. Anyone with the `dataquery_view` permission has access to the DQT, and therefore has access to all data in the LORIS instance. 
```
Let's leave these unresolved so that the original change-requester (me in this case) can verify - (i.e. leave them a little time to resolve the conversation)

christinerogers commented:
Thanks for reverting @knoll-alex 👍 Let's also find a way next week to revert the [second-last change-commit `0b4f74`])https://github.com/aces/Loris/pull/5206/commits/0b4f74c59e8a3f08327e6929365575396719f95b)

driusan commented on modules/dataquery/readme.md:
````diff
@@ -0,0 +1,79 @@
+    <admin>adminuser</admin>
+    <adminpass>adminpass</adminpass>
+</CouchDB>
+```
````
That doesn't exist anymore, the `<SyncAccounts>` part should be removed.

knoll-alex replied:
Removed!

knoll-alex replied:
Removed

christinerogers replied:
resolving 👍

driusan commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,75 @@
+## Configuration
+Use of the DQT module requires setting up the CouchDB server and credentials:
+
+[CouchDB](http://couchdb.apache.org)
```
This should also link to https://github.com/aces/Data-Query-Tool as a requirement, I think.

knoll-alex replied:
Isn't that link just a repeat of the info laid out in this readme?

knoll-alex replied:
Isn't everything in that link addressed in this readme?

christinerogers replied:
It may be less than 100% up-to-date but still contains tips for developers (vs. those just installing the end product)

knoll-alex replied:
gotcha, thanks! I added a "note" at the end of config.

christinerogers replied:
👍

knoll-alex commented:
@driusan @johnsaigle @christinerogers Thanks for everyone's patience with this PR - would appreciate a final review to ensure I've addressed all comments so we can MERGE!

johnsaigle commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,76 @@
+## Permissions
+Users must have `dataquery_view` permission to use the DQT. 
+
+It’s important to note that more granular permissions do not exist for this module, to restrict access to a subset of the data. Anyone with the `dataquery_view` permission has access to view, query and download data from the DQT, and therefore has access to all data loaded in the DQT from this LORIS instance.
```
```suggestion
It’s important to note that granular permissions do not exist for this module. Anyone with the `dataquery_view` permission has access to view, query and download data from the DQT, and therefore has access to all data loaded in the DQT from this LORIS instance. It is not possible to restrict user access to a subset of the data.
```

Small grammar improvement.

johnsaigle commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,76 @@
+It’s important to note that more granular permissions do not exist for this module, to restrict access to a subset of the data. Anyone with the `dataquery_view` permission has access to view, query and download data from the DQT, and therefore has access to all data loaded in the DQT from this LORIS instance.
+
+## Interactions with LORIS
+Before the DQT can be used, the CouchDB import scripts must be run to load data from LORIS' MySQL tables into CouchDB. These scripts are found in the `tools/` directory, named `CouchDB_Import_*.php`.
```
```suggestion
Before the DQT can be used, the CouchDB import scripts must be run in order to load data from LORIS's MySQL tables into CouchDB. These scripts are found in the `tools/` directory, named `CouchDB_Import_*.php`.
```

christinerogers replied:
Agree - @knoll-alex could you commit this suggested tweak?

johnsaigle commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,76 @@
+
+#### NOTE - for developer setup tips, see further instructions [here](https://github.com/aces/Data-Query-Tool)
+
+    <port>5984</port>
```
I'm not quite sure why this line is here.

knoll-alex commented:
Requested changed have been made. I just tested the markdown text using the Help Editor in the LORIS demo, to test how it would render in the front end. It seemed to have worked back in September, but now it's not rendering properly. Screenshot below. Does anyone know why?

![Screenshot 2019-11-01 18 03 50](https://user-images.githubusercontent.com/37813606/68059251-3429d800-fcd2-11e9-93dc-84de0f9b5090.png)

christinerogers commented:
Not sure what to suggest about the hash tags. 
My first instinct for troubleshooting is -- Can you point us to an example that's working successfully with that level of header tag, and we'll go over it carefully?   maybe in a different module?

knoll-alex commented:
> Not sure what to suggest about the hash tags.
> My first instinct for troubleshooting is -- Can you point us to an example that's working successfully with that level of header tag, and we'll go over it carefully? maybe in a different module?

Sure - one good example I found is the data team helper. Here's a link to it's raw text: https://raw.githubusercontent.com/aces/Loris/major/modules/data_team_helper/help/data_team_helper.md

and here's a screenshot of how it renders in LORIS:

![Screenshot 2019-11-06 17 32 54](https://user-images.githubusercontent.com/37813606/68343743-7df03500-00bb-11ea-8613-710a7915049b.png)

knoll-alex commented:
Nevermind!! I was missing spaces after each heading. I fixed it and it's rendering fine now. I'm going to update the headers to make them smaller though

![Screenshot 2019-11-06 17 34 40](https://user-images.githubusercontent.com/37813606/68343919-eb03ca80-00bb-11ea-9aa3-bbc15b31fcdd.png)

christinerogers commented:
OK great.  keep the screenshots coming so we can approve this efficiently. 

pro tip:  go back and remove older screenshots that are no longer relevant -- decreases the scrolling action  and confusion :)

knoll-alex commented:
Just realized... this is a readme so it doesn't matter how it renders in LORIS! Right?

christinerogers commented:
rebased to 22 branch 👍

johnsaigle commented on modules/dataquery/readme.md:
```diff
@@ -0,0 +1,85 @@
+
+## Purpose
+
+The Data Query Tool (“The DQT”) is used to query and export data stored in LORIS. Designed as a front-end tool, it is located within the Reports drop-down menu. 
```
```suggestion
The Data Query Tool (“The DQT”) is used to query and export data stored in LORIS. It is located within the Reports drop-down menu. 
```

christinerogers replied:
makes sense. 👍

johnsaigle commented on modules/dataquery/readme.md:
````diff
@@ -0,0 +1,85 @@
+
+3. Amend the `<CouchDB>` section of your LORIS `project/config.xml` as follows:
+
+```
````
```suggestion
```xml
```

This should make the formatting a little nicer when rendered in GitHub.

christinerogers commented:
@driusan This module spec should be ready to merge (for the 22 release), pending your re-review/merge.
